src/mono/mono/metadata/custom-attrs.c

```diff
@@ -1332,7 +1553,8 @@ mono_reflection_create_custom_attr_data_args_noalloc (MonoImage *image, MonoMeth
 			arginfo [j].type = field->type;
```

Comment:
Maybe change the typed_args_out and named_args_out
parameters of mono_reflection_create_custom_attr_data_args_noalloc ()
to MonoCustomAttrValue** so the callers don't have to cast them.

Alternatively, add a MonoDecodeCustomAttr or some other type which contains
everything else, and add a free function so the caller doesn't have to free everything manually.

Reply:
@vargaz does this also mean that eventually we will have to update the managed variant in the same fashion?

Reply:
The managed variant allocates objects and arrays, so there is less to free there.

Reply:
`MonoDecodeCustomAttr` introduced with a free function used by callers